File: app/src/main/java/lemond/annoying/gamerscompanion/core/viewmodel/ViewMoreViewModel.java
package lemond.annoying.gamerscompanion.core.viewmodel;


import android.view.View;
import android.widget.Toast;

public class ViewMoreViewModel {

    public enum ViewMoreType {
        NEWS,
        HYPED_GAMES,
        POPULAR_GAMES,
        TRENDING_GAMES
    }

    private final ViewMoreType type;
    private final String viewMoreText;

    public ViewMoreViewModel(ViewMoreType type, String viewMoreText) {
        this.type = type;
        this.viewMoreText = viewMoreText;
    }

    public String getViewMoreText() {
        return viewMoreText;
    }

    public void onViewMoreClick(View view) {
        switch (type) {
            case HYPED_GAMES:
                break;
            case POPULAR_GAMES:
                break;
            case TRENDING_GAMES:
                break;
            case NEWS:
                // TODO: 2018-02-16 launch view more news activity here
                Toast.makeText(view.getContext(), "to be implemented", Toast.LENGTH_SHORT).show();
                break;
        }
    }
}

File: app/src/main/java/lemond/annoying/gamerscompanion/core/viewmodel/ViewMoreViewModel.kt
package lemond.annoying.gamerscompanion.core.viewmodel


import android.view.View
import android.widget.Toast

class ViewMoreViewModel(private val type: ViewMoreType, val viewMoreText: String) {

    enum class ViewMoreType {
        NEWS,
        HYPED_GAMES,
        POPULAR_GAMES,
        TRENDING_GAMES
    }

    fun onViewMoreClick(view: View) {
        when (type) {
            ViewMoreViewModel.ViewMoreType.HYPED_GAMES -> {
            }
            ViewMoreViewModel.ViewMoreType.POPULAR_GAMES -> {
            }
            ViewMoreViewModel.ViewMoreType.TRENDING_GAMES -> {
            }
            ViewMoreViewModel.ViewMoreType.NEWS ->
                // TODO: 2018-02-16 launch view more news activity here
                Toast.makeText(view.context, "to be implemented", Toast.LENGTH_SHORT).show()
        }
    }
}
